src/launchpad/analyzers/ios.py

```diff
@@ -57,53 +55,70 @@ def __init__(
         self.enable_range_mapping = enable_range_mapping
         self.enable_treemap = enable_treemap
         self._temp_dirs: List[Path] = []
+        self.binary_analysis: IOSBinaryAnalysis | None = None
 
     def analyze(self, input_path: Path) -> IOSAnalysisResults:
         """Analyze an iOS app bundle.
 
         Args:
-            input_path: Path to zip archive
+            input_path: Path to the .xcarchive.zip file
 
         Returns:
-            Complete analysis results
-
-        Raises:
-            ValueError: If input is not a valid iOS app bundle
-            RuntimeError: If analysis fails
+            Analysis results including file sizes, binary analysis, and treemap
         """
-        logger.info(f"Starting iOS analysis of {input_path}")
-
-        analysis_start_time = time.time()
+        start_time = time.time()
+        logger.info(f"Analyzing iOS app: {input_path}")
 
-        # TODO: Add vaildation to ensure is xcarchive (or support other formats)
-        artifact = ZippedXCArchive(input_path.read_bytes())
+        with open(input_path, "rb") as f:
+            xcarchive = ZippedXCArchive(f.read(), self.working_dir)
+        self._temp_dirs.append(xcarchive.extract_dir)
 
-        # Extract basic app information
-        app_info = self._extract_app_info(artifact)
-        logger.info(f"Analyzing app: {app_info.name} v{app_info.version}")
+        app_info = self._extract_app_info(xcarchive)
 
-        # Analyze files in the bundle
-        file_analysis = self._analyze_files(artifact)
-        logger.info(f"Found {file_analysis.file_count} files, " f"total size: {file_analysis.total_size} bytes")
+        file_analysis = self._analyze_files(xcarchive)
+        treemap = None
+        binary_analysis: List[IOSBinaryAnalysis] = []
 
-        # Generate treemap if enabled
-        treemap_results = None
         if self.enable_treemap:
-            treemap_results = self._generate_treemap(app_info, file_analysis)
-            logger.info(f"Generated treemap with {treemap_results.file_count} files")
-
-            # Analyze the main executable binary
-        binary_analysis = self._analyze_binary(artifact)
-        logger.info(f"Binary analysis complete, " f"executable size: {binary_analysis.executable_size} bytes")
+            treemap_builder = TreemapBuilder(
+                app_name=app_info.name,
+                platform="ios",
+                download_compression_ratio=0.8,  # iOS apps typically compress to ~80%
+            )
+            treemap = treemap_builder.build_file_treemap(file_analysis)
 
-        return IOSAnalysisResults(
+            # If range mapping is enabled, add binary section details to treemap
+            if self.enable_range_mapping:
```

Comment:
I might consolidate some of these flags later on since it's already getting a bit confusing, but it's nice having size features toggleable